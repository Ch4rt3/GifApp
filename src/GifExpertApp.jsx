import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch']);

    const onAddCategory = (newCategory) => {

        if(categories.includes(newCategory)) return;

        setCategories([newCategory,...categories]);
    }

  return (
    <>
        <h3>GifExpertApp</h3>
        <br/>


        <AddCategory 
            onNewCategory={ onAddCategory }
        />


        <ol>
            { categories.map( category => {
                return(
                    <GifGrid
                        key={category}
                        category={category}
                    />
                )
            }) 
            }
            
        </ol>
    </>
  )
}
