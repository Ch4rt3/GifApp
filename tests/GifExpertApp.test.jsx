import { fireEvent, screen } from "@testing-library/dom";
import { render } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";


describe('pruebas en <GifExpertApp />', () => {

    test('Debe hacer match con el snapshot', () => { 
        
        const { container } = render( <GifExpertApp/> );

        expect( container ).toMatchSnapshot();

     });
    
    test('Debe mostrarse el titulo GifExpertApp', () => { 
        
        render(<GifExpertApp/>);

        expect(screen.getByText("GifExpertApp")).toBeTruthy();

     });
    
    test('Debe agregar una nueva categoria', () => { 
        
        const newCategory = 'Fortnite';

        render(<GifExpertApp/>);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input( input, { target: { value: newCategory } });
        fireEvent.submit( form );

        expect( screen.getByText(newCategory )).toBeTruthy();


     });
    
    test('No debe agregar una categoria ya existente ', () => { 
        
        const categoria = 'Valorant';

        render(<GifExpertApp/>);

        const input = screen.getByRole('textbox');
        const form = screen.getByRole('form');

        fireEvent.input(input, { target : { value: categoria } });
        fireEvent.submit(form);
        fireEvent.input(input, { target : { value: categoria } });
        fireEvent.submit(form);

        expect(screen.getAllByText(categoria).length).toBe(1);

     });

 });
