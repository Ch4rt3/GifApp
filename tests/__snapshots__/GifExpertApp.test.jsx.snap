// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h3>
    GifExpertApp
  </h3>
  <br />
  <form
    aria-label="form"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
  <ol>
    <h3>
      One Punch
    </h3>
    <h2>
      Cargando...
    </h2>
    <div
      class="card-grid"
    />
  </ol>
</div>
`;
